packages/apollo-server-fastify/src/ApolloServer.ts

```diff
@@ -170,11 +170,17 @@ export class ApolloServer extends ApolloServerBase {
             beforeHandlers.push(fileUploadMiddleware(this.uploadsConfig, this));
           }
 
+          const graphQLServerOptions = this.graphQLServerOptions.bind(this);
+          const applyContextArgs = async (
+            request?: FastifyRequest<IncomingMessage>,
+            reply?: FastifyReply<OutgoingMessage>,
+          ) => graphQLServerOptions({ request, reply });
```

Comment:
For symmetry reasons, I think we should make the Fastify integration have define (almost exactly this, except with an explicit return type of `Promise<GraphQLOptions>`) as a `createGraphQLServerOptions` class method [like the other integrations](https://github.com/apollographql/apollo-server/search?q=%22createGraphQLServerOptions%22&unscoped_q=%22createGraphQLServerOptions%22) in the (this) Apollo Server repository.  Then, calling `super.graphQLServerOptions({ request, reply })` within that function to satisfy the return type.